04_numeronym/shahriys/main_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	var bufnum bytes.Buffer
+	outnum = &bufnum
+
+	main()
+	expected := `[a11y K8s abc]`
+	actual := bufnum.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()" + expected + " " + actual)
+	}
+
+}
+func TestNumeronym(t *testing.T) {
+
+	type test struct {
+		expected []string
+		actual   []string
```

Comment:
Normally you would only store the input and expected, not actual, rather than evaluating the function while creating the test data.
This is because often times you test functions that can return multiple values, you can't capture multiple return values in an inline struct initialization.
I suggest using an input slice, and evaluating the result in the test before testing it. Remember to call with
`numeronyms(tc.input...)`